resources/functionMap.php

```diff
@@ -10547,13 +10547,13 @@
 'snmp3_walk' => ['array|false', 'host'=>'string', 'sec_name'=>'string', 'sec_level'=>'string', 'auth_protocol'=>'string', 'auth_passphrase'=>'string', 'priv_protocol'=>'string', 'priv_passphrase'=>'string', 'object_id'=>'string', 'timeout='=>'int', 'retries='=>'int'],
 'SNMP::__construct' => ['void', 'version'=>'int', 'hostname'=>'string', 'community'=>'string', 'timeout='=>'int', 'retries='=>'int'],
 'SNMP::close' => ['bool'],
-'SNMP::get' => ['array|string|false', 'object_id'=>'string|array', 'preserve_keys='=>'bool'],
+'SNMP::get' => ['($objectId is array ? array<string,string> : string)|false', 'objectId'=>'string|string[]', 'preserveKeys='=>'bool'],
```

Comment:
I'm not sure conditional type will work here in functionMap. Please add a test in `tests/PHPStan/Analyser/nsrt` to verify that.

Reply:
Correct, I should not have assumed it would. I updated to use the multiple signature syntax, I also provided more specificity for string parameters and updated the function-based calls as well.